src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
```

Comment:
You can create separate function which will do this code - people.some(m => m.name === p.mother))
Remember main clean-code rule - Don't Repeat Yourself.

If smth occurs twice or more - separate it. Follow this rule in the entire file